spyder/plugins/ipythonconsole/widgets/kernelconnect.py

```diff
@@ -205,6 +217,29 @@ def load_connection_settings(self):
         except Exception:
             pass
 
+    def _validate_connection_file(self):
+        cf_path = osp.dirname(self.cf.text())
+        cf_filename = osp.basename(self.cf.text())
+        try:
+            if not cf_filename.endswith(".json"):
+                cf_filename += ".json"
+            connection_file = find_connection_file(
+                filename=cf_filename, path=cf_path
+            )
+        except OSError:
+            connection_file = None
+
+        if connection_file is None:
+            QMessageBox.warning(
+                self,
+                _('Warning'),
+                _("The value for the connection file is not correct."),
```

Comment:
```suggestion
            QMessageBox.error(
                self,
                _('Error'),
                _(
                    "The connection file you passed doesn't exist or can't be "
                    "found by Spyder."
                ),
```

* Let's show an error here instead of a warning.
* Improve message we show to users.